tests/api/ocsp.c

```diff
@@ -0,0 +1,539 @@
+#include "ocsp.h"
+#include "ocsp_test_blobs.h"
+#include <tests/unit.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/ocsp.h>
+#include <wolfssl/ssl.h>
+
+#if defined(HAVE_OCSP)
+struct ocsp_cb_ctx {
+    byte* response;
+    int responseSz;
+};
+
+struct test_conf {
+    unsigned char* resp;
+    int respSz;
+    unsigned char* ca0;
+    int ca0Sz;
+    unsigned char* ca1;
+    int ca1Sz;
+    unsigned char* targetCert;
+    int targetCertSz;
+};
+
+static int ocsp_cb(void* ctx, const char* url, int urlSz, unsigned char* req,
+    int reqSz, unsigned char** respBuf)
+{
+    struct ocsp_cb_ctx* cb_ctx = (struct ocsp_cb_ctx*)ctx;
+    (void)url;
+    (void)urlSz;
+    (void)req;
+    (void)reqSz;
+
+    *respBuf = cb_ctx->response;
+    return cb_ctx->responseSz;
+}
+
+static int test_ocsp_response_with_cm(struct test_conf* c)
+{
+    EXPECT_DECLS;
+    WOLFSSL_CERT_MANAGER* cm = NULL;
+    struct ocsp_cb_ctx cb_ctx;
+    int ret;
+
+    cm = wolfSSL_CertManagerNew();
+    ExpectPtrNE(cm, NULL);
+    ret = wolfSSL_CertManagerEnableOCSP(cm,
+        WOLFSSL_OCSP_URL_OVERRIDE | WOLFSSL_OCSP_NO_NONCE);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    ret = wolfSSL_CertManagerSetOCSPOverrideURL(cm, "http://foo.com");
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    cb_ctx.response = (byte*)c->resp;
+    cb_ctx.responseSz = c->respSz;
+    ret = wolfSSL_CertManagerSetOCSP_Cb(cm, ocsp_cb, NULL, (void*)&cb_ctx);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    /* add ca in cm */
+    if (c->ca0 != NULL) {
+        ret = wolfSSL_CertManagerLoadCABuffer(cm, c->ca0, c->ca0Sz,
+            WOLFSSL_FILETYPE_ASN1);
+        ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    }
+    if (c->ca1 != NULL) {
+        ret = wolfSSL_CertManagerLoadCABuffer(cm, c->ca1, c->ca1Sz,
+            WOLFSSL_FILETYPE_ASN1);
+        ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    }
+    /* check cert */
+    ret = wolfSSL_CertManagerCheckOCSP(cm, c->targetCert, c->targetCertSz);
+    wolfSSL_CertManagerFree(cm);
+    return ret;
+}
```

Comment:
Either use the `Expect` API fully or not. This is confusing. You can add an `expectedRet` param for the last check.